discovery-provider/src/tasks/user_library.py

```diff
@@ -94,6 +96,11 @@ def user_library_state_update(
 
     return num_total_changes
 
+######## HELPERS ########
+
+def dispatch_favorite(session, bus, save, block_number):
+    user_id = save.user_id
+    bus.dispatch(session, ChallengeEvent.favorite, block_number, user_id)
```

Comment:
NIT - why do we need to assign user_id as a var instead of just having the arg be save.user_id...?

Reply:
no good reason, I think I just thought this was a bit cleaner. but can lose it